### PR TITLE
[DOC] Fix example following comments in  issue #6865

### PR DESCRIPTION
**Pull request by rougier**
I fixed the example by extracting the relevant code from [geo.py](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/projections/geo.py) and copying back comments from the old example.

### CODE REVIEW

File: examples/api/custom_projection_example.py [hunk 1]
```diff
@@ -9,73 +9,69 @@
...
-    projection.
-
-    http://en.wikipedia.org/wiki/Hammer_projection
+    An abstract base class for geographic projections
     """
-    # The projection must specify a name.  This will be used be the
-    # user to select the projection, i.e. ``subplot(111,
-    # projection='custom_hammer')``.
-    name = 'custom_hammer'
+    class ThetaFormatter(Formatter):
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: I think you can import this from `matplotlib.projections.polar.ThetaFormatter`.

File: examples/api/custom_projection_example.py [hunk 2]
```diff
@@ -271,8 +268,8 @@ def format_coord(self, lon, lat):
 
         In this case, we want them to be displayed in degrees N/S/E/W.
         """
-        lon = np.degrees(lon)
-        lat = np.degrees(lat)
+        lon = lon * (180.0 / np.pi)
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: Why change this?

File: examples/api/custom_projection_example.py [hunk 3]
```diff
@@ -346,13 +320,11 @@ def set_longitude_grid_ends(self, degrees):
         class -- it provides an interface to something that has no
         analogy in the base Axes class.
         """
-        longitude_cap = np.radians(degrees)
-        # Change the xaxis gridlines transform so that it draws from
-        # -degrees to degrees, rather than -pi to pi.
+        self._longitude_cap = degrees * (np.pi / 180.0)
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: Why change this?

File: examples/api/custom_projection_example.py [hunk 4]
```diff
@@ -380,7 +360,19 @@ def end_pan(self):
...
         pass
 
-    # Now, the transforms themselves.
+
+class HammerAxes(GeoAxes):
+    """
+    A custom class for the Aitoff-Hammer projection, an equal-area map
+    projection.
+
+    http://en.wikipedia.org/wiki/Hammer_projection
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: https

File: examples/api/custom_projection_example.py [hunk 5]
```diff
@@ -380,7 +360,19 @@ def end_pan(self):
...
+    """
+
+    # The projection must specify a name.  This will be used be the
```
Comment (QuLogic) [inline examples/api/custom_projection_example.py]: s/be the/by the/

File: examples/api/custom_projection_example.py (hunk 1 above)
Comment (tacaswell) [inline examples/api/custom_projection_example.py]: I think part of the exercise here is to write one mostly from scratch.

File: examples/api/custom_projection_example.py (hunk 3 above)
Comment (rougier) [inline examples/api/custom_projection_example.py]: Cause it is how it is writtern in geo.py

File: examples/api/custom_projection_example.py (hunk 2 above)
Comment (rougier) [inline examples/api/custom_projection_example.py]: Cause it is how it is writtern in geo.py

File: examples/api/custom_projection_example.py (hunk 4 above)
Comment (rougier) [inline examples/api/custom_projection_example.py]: ?

**Comment by tacaswell** [comment]
backported to v2.x as 07c8506 

Thanks @rougier !  Your work on these examples is _super_ appreciated!

**Pull request merged**